### PR TITLE
Dockerize eints

### PR DESCRIPTION
**Pull request by TrueBrain**
In the "run" file are a lot of help strings I really did not know to answer. Hopefully someone can fill in the gaps.

It only supports `development`, `github`, and `ldap` authentication. `redmine` is not implemented, as it would bloat the dependencies drastically.

### CODE REVIEW

File: run [hunk 1]
```diff
@@ -1,53 +1,118 @@
...
+@click.option(
+    "--authentication",
+    help="Authentication protocol to use",
+    type=click.Choice(["development", "github"], case_sensitive=False),
+    default="development",
+)
+@click.option("--stable-languages", help="Folder with stable languages", default="stable_languages")
+@click.option("--unstable-languages", help="Folder with unstable languages", default="unstable_languages")
+@click.option("--project-root", help="Root folder for data storage", required=True)
+@click.option("--project-cache", help="I have no idea", default=1)
```
Comment (frosch123) [inline run:29]: Number of projects kept in memory LRU cache.

File: run [hunk 2]
```diff
@@ -1,53 +1,118 @@
...
+@click.option(
+    "--project-types",
+    help="I have no idea",
```
Comment (frosch123) [inline run:32]: Space separated list of allowed project types.

File: run [hunk 3]
```diff
@@ -1,53 +1,118 @@
...
+    multiple=True,
+    type=click.Choice(["newgrf", "game-script", "openttd"], case_sensitive=False),
```
Comment (frosch123) [inline run:34]: Does "Choice" allow listing multiple types? This parameter is more like a "set".

File: run [hunk 4]
```diff
@@ -1,53 +1,118 @@
...
+    default=["newgrf", "game-script"],
+)
+@click.option(
+    "--storage-format",
+    help="No clue",
```
Comment (frosch123) [inline run:39]: Storage format for project. Either one huge file per project, or one folder per project with files per language.

File: run [hunk 5]
```diff
@@ -1,53 +1,118 @@
...
+    type=click.Choice(["one-file", "split-languages"], case_sensitive=False),
+    default="one-file",
+)
+@click.option(
+    "--data-format",
+    help="Format to store project data in",
+    type=click.Choice(["xml", "json"], case_sensitive=False),
+    default="xml",
+)
+@click.option("--language-file-size", help="really, wuth?", default=100000)
```
Comment (frosch123) [inline run:49]: Uploads larger than this are rejected. I guess this should be a nginx setting :)

File: run [hunk 6]
```diff
@@ -1,53 +1,118 @@
...
+@click.option("--num-backup-files", help="How many backup files for project data to keep", default=5)
+@click.option("--max-num-changes", help="Sometimes important I am sure", default=5)
```
Comment (frosch123) [inline run:51]: Length of string history to keep.

File: run [hunk 7]
```diff
@@ -1,53 +1,118 @@
...
+@click.option("--min-num-changes", help="another very important flag", default=2)
```
Comment (frosch123) [inline run:52]: Yeah, I don't know either... I know it must be >= 2 or the git scripts fail.
Maybe hardcode to 2 here?

File: run [hunk 8]
```diff
@@ -1,53 +1,118 @@
...
+@click.option("--change-stable-age", help="The age of the world, or something", default=600)
```
Comment (frosch123) [inline run:53]: No idea either. I could give it some sense if I turned a "<" into a ">"...

**Comment by frosch123** [comment]
Maybe put "look at docs/manual/setup.rst" in case we do not understand :)

File: run (hunk 3 above)
Comment (TrueBrain) [reply run:34]: I will cast it to a set first, so it behaves like a set :)

**Review by frosch123 — APPROVED** [review]

**Pull request merged**